generator/build_integration.zig

```diff
@@ -4,42 +4,24 @@ const Builder = std.build.Builder;
 const Step = std.build.Step;
 const GeneratedFile = std.build.GeneratedFile;
 
-/// Stage the shader should be built for. This is passed to the -fshader-stage
-/// argument when invoking glslc.
-pub const ShaderStage = enum {
-    vertex,
-    fragment,
-    tesscontrol,
-    tesseval,
-    geometry,
-    compute,
-};
-
 /// Utility functionality to help with compiling shaders from build.zig.
-/// Invokes glslc (or another shader compiler passed to `init`) for each shader
+/// Invokes a shader compile command (e.g., glslc ...) for each shader
 /// added via `addShader`.
 pub const ShaderCompileStep = struct {
     /// The directory within the zig-cache directory that is used to store
     /// shader artifacts.
     pub const cache_dir = "shaders";
 
-    /// This structure contains additional options that can be passed to glslc when shaders are compiled.
+    /// This structure contains additional options that pertain to specific shaders only.
     pub const ShaderOptions = struct {
```

Comment:
maybe at this point it just makes sense to get rid of the struct and have the function take additional args directly